### PR TITLE
Added function to format execution time

### PR DESCRIPTION
**Pull request by lisandrasilva**
Since the pretty-print of the execution time is being used in multiple places, this PR introduces a function that returns the `str` with the formatted time.

Addresses this [comment](https://github.com/runtimeverification/kontrol/pull/391#discussion_r1510546009) in Kontrol.

### CODE REVIEW

File: src/pyk/proof/reachability.py [hunk 1]
```diff
@@ -784,7 +797,7 @@ def lines(self) -> list[str]:
             f'    terminal: {self.terminal}',
             f'    refuted: {self.refuted}',
             f'    bounded: {self.bounded}',
-            f'    execution time: {self.exec_time // 3600}h {(self.exec_time % 3600) // 60}m {self.exec_time % 60}s',
+            f'    execution time: {self.exec_time}',
```
Comment (palinatolmach) [inline src/pyk/proof/reachability.py:800]: Should this be `self.formatted_exec_time()`?
```suggestion
            f'    execution time: {self.formatted_exec_time()}',
```

**Review by palinatolmach — COMMENTED** [review]
Left one comment, the rest looks good to me!

File: src/pyk/proof/reachability.py (hunk 1 above)
Comment (lisandrasilva) [reply src/pyk/proof/reachability.py:800]: This is a parameter for the `APRSummary` which I called `exec_time : str`, but I can call it `formatted_exec_time`, maybe it makes more sense :)

**Review by palinatolmach — APPROVED** [review]

File: src/pyk/proof/reachability.py [hunk 2]
```diff
@@ -213,6 +213,19 @@ def add_exec_time(self, exec_time: float) -> None:
     def set_exec_time(self, exec_time: float) -> None:
         self._exec_time = exec_time
 
+    def formatted_exec_time(self) -> str:
+        exec_time = round(self.exec_time)
+        h = exec_time // 3600
+        m = (exec_time % 3600) // 60
+        s = exec_time % 60
```
[the remark this answers is not in the document]
Comment (lisandrasilva) [reply src/pyk/proof/reachability.py]: Thank you!

File: src/pyk/proof/reachability.py [hunk 3]
```diff
@@ -213,6 +213,19 @@ def add_exec_time(self, exec_time: float) -> None:
...
+        if h != 0:
+            return f'{h}h {m}m {s}s'
+        else:
+            if m != 0:
+                return f'{m}m {s}s'
+            else:
+                return f'{s}s'
```
[the remark this answers is not in the document]
Comment (lisandrasilva) [reply src/pyk/proof/reachability.py]: Yes, thank you for the suggestion :)

**Review by anvacaru — APPROVED** [review]

**Pull request merged**